### PR TITLE
[bitnami/prometheus-operator] Expose additional ports from Thanos sidecar

### PR DESCRIPTION
**Pull request by hwoarang**
The sidecar is configured to expose the following two ports

- GRPC/10901
- HTTP/10902

However, the service that targets the sidecar does not expose the later
making it impossible to get any metrics from it through traditional
prometheus scraping as these are only available to the HTTP endpoint.
As such, add a new option to allow us to expose additional ports from
the sidecar container.

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`hart/kubeapps). This is only a synchronized mirror.
- [x] Variables are documented in the README.md
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml [hunk 1]
```diff
@@ -29,6 +29,10 @@ spec:
       {{- if and .Values.prometheus.thanos.service.nodePort (or (eq .Values.prometheus.thanos.service.type "NodePort") (eq .Values.prometheus.thanos.service.type "LoadBalancer")) }}
       nodePort: {{ .Values.prometheus.thanos.service.nodePort }}
       {{- end }}
+    - name: http
+      port: 10902
+      targetPort: http
+      protocol: TCP
```
Comment (dani8art) [inline bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml:35]: I would do it customizable as grpc port is with `.Values.prometheus.thanos.service.port`. In fact, you could refactor this values:

```yaml
prometheus:
  thanos:
    service:
      ports:
        grpc: 10901
        http: 10902
```

then use them
```
.Values.prometheus.thanos.service.ports.grpc
.Values.prometheus.thanos.service.ports.http
```

File: bitnami/prometheus-operator/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 0.40.0
 description: The Prometheus Operator for Kubernetes provides easy monitoring definitions for Kubernetes services and deployment and management of Prometheus instances.
 name: prometheus-operator
-version: 0.22.3
+version: 0.22.4
```
Comment (dani8art) [inline bitnami/prometheus-operator/Chart.yaml:5]: I think this should be a minor bump

File: bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml (hunk 1 above)
Comment (hwoarang) [reply bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml:35]: @dani8art that would require a migration path for existing users (and as a result a major bump in Chart version). That's mainly the reason why I did not do it like this.

File: bitnami/prometheus-operator/Chart.yaml (hunk 2 above)
Comment (hwoarang) [reply bitnami/prometheus-operator/Chart.yaml:5]: @dani8art if we follow your approach then we will need a major bump as the change is breaking backwards compatibility and users need to perform manual steps to migrate no?

File: bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml (hunk 1 above)
Comment (dani8art) [reply bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml:35]: Yes you are right, and what about the `Values.prometheus.thanos.service.extraPorts` approach? please take a look at this https://github.com/bitnami/charts/blob/b0c0732b5ef07d31a84cb64d2770b34fee6ecc57/bitnami/discourse/templates/service.yaml#L36

Comment (dani8art) [reply bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml:35]: then you could set it by default in values:

```yaml
prometheus:
  thanos:
    service:
      extraPorts:
        - name: http
          port: 10902
          targetPort: http
          protocol: TCP
```

Comment (hwoarang) [reply bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml:35]: @dani8art that sounds like a very good suggestion. Thanks

Comment (hwoarang) [reply bitnami/prometheus-operator/templates/prometheus/thanos-service.yaml]: It's done

File: bitnami/prometheus-operator/values-production.yaml [hunk 3]
```diff
@@ -692,6 +692,15 @@ prometheus:
...
       annotations: {}
 
+      ## Extra ports to expose from the Thanos sidecar container
+      ##
+      # extraPorts:
+      #   - name: http
+      #     port: 10902
+      #     targetPort: http
+      #     protocol: TCP
+      extraPorts: {}
```
Comment (dani8art) [inline bitnami/prometheus-operator/values-production.yaml:702]: it should be an array, please change it to `[]`

**Pull request merged**